controllers/front/action.php

```diff
@@ -17,25 +18,366 @@
+use PrestaShop\PrestaShop\Core\Product\ProductPresentationSettings;
+
+
 class BlockWishlistActionModuleFrontController extends ModuleFrontController
 {
     public function initContent()
     {
         parent::initContent();
 
+        if (!$this->context->customer->isLogged()) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('You aren\'t logged in', 'mywishlist'),
+                ])
+            );
+        }
+
         $params = Tools::getValue('params');
 
+        $params = [
+            'idWishlist' => 3,
+            'id_product' => 1,
+            'id_product_attribute' => 3,
+            'quantity' => 2,
+            'id_cart' => 25,
+            'name' => 'newWishTestName'
+        ];
+        /*
+        Here we call all methods dinamically given by the path
+        */
         if (method_exists($this, Tools::getValue('action') . 'Action')) {
             call_user_func([$this, Tools::getValue('action') . 'Action'], $params);
         }
 
-        dump('here');
-        die;
-        // return error response
+        $this->ajaxDie(
+            Tools::jsonEncode([
+                'success' => false,
+                'message' => $this->module->l('Unknow action', 'mywishlist'),
+            ])
+        );
+    }
+
+    public function addProductToWishlistAction($params)
+    {
+        $id_customer = (int) $this->context->customer->id;
+        $idWishlist = (int) $params['idWishlist'];
+        $id_product = (int) $params['id_product'];
+        $id_product_attribute = (int) $params['id_product_attribute'];
+        $quantity = (int) $params['quantity'];
+
+        if ($idWishlist !== 0) {
+            if (wishlist::exists($idWishlist, $this->context->customer->id) === false) {
+                $wishlist = new wishlist();
+                $wishlist->id_shop = $this->context->shop->id;
+                $wishlist->id_shop_group = $this->context->shop->id_shop_group;
+                $wishlist->id_customer = $this->context->customer->id;
+                $wishlist->name = 'default';
+                $wishlist->token = $this->generateWishlistToken();
+                $wishlist->default = 1;
+                $wishlist->add();
+            }
+        } else {
+            $wishlist = new Wishlist($idWishlist);
+        }
+
+        $productIsAdded = $wishlist->addProduct(
+            $idWishlist,
+            $this->context->customer->id,
+            $id_product,
+            $id_product_attribute,
+            $quantity
+        );
+
+        if ($productIsAdded === false) {
+            $this->ajaxDie(
+                Tools::jsonEncode([
+                    'success' => false,
+                    'message' => $this->module->l('There was an error adding the product', 'mywishlist'),
+                ])
+            );
+        }
+
+        $this->ajaxDie(
+            Tools::jsonEncode([
```

Comment:
```suggestion
            json_encode([
```
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/classes/Tools.php#L3007